inspirehep/modules/records/tasks.py

```diff
@@ -0,0 +1,136 @@
+from celery.utils.log import get_task_logger
+from elasticsearch.helpers import scan
+
+from invenio_search import current_search_client as es
+from invenio_db import db
+from invenio_records.api import Record
+
+from inspire_schemas import api
+from inspirehep.modules.search.utils import dotter
+
+logger = get_task_logger(__name__)
+
+bad_fields = ['deleted_records', 'new_record', 'self']
+
+
+@shared_task(ignore_result=True)
+def replace_old_ref_for_record(old_record_ref, new_record_ref):
+    possible_fields = get_ref_fields()
+    record_uuids = es_search_pointing_ids_from_ref_in_terms(possible_fields, old_record_ref)
+
+    for rec_uuid in record_uuids:
+        rec_db = Record.get_record(rec_uuid)
+        find_and_replace_old_link_with_new_link_in_record(rec_db, new_record_ref, old_record_ref)
+        rec_db.commit()
+    db.session.commit()
+
+
+def remove_dot(string):
+    return string[1:]
+
+
+def get_recid_from_url(url):
+    try:
+        res = int(url.split('/')[-1])
+    except ValueError:
+        res = None
+    return res
+
+
+def find_ref_fields_in_schema(collection, fields):
+    fields_dotted = []
+
+    schema = api.load_schema(collection)
+    all_fields_dotted = dotter(schema.get('properties'), '', [])
+    for field_dotted in all_fields_dotted:
+        if '$ref' in field_dotted:
+            if not any(field in field_dotted for field in fields):
+                if not any(bad_field in field_dotted for bad_field in bad_fields):
+                    fields_dotted.append(field_dotted)
+
+    return [remove_dot(str(field)) for field in fields_dotted]
+
+
+def get_ref_fields(collections=None):
+    if collections is None:
+        collections = ['hep', 'authors', 'conferences',
+                       'experiments', 'institutions',
+                       'jobs', 'journals']
+    fields = []
+
+    for collection in collections:
+        fields += find_ref_fields_in_schema(collection, fields)
+
+    return fields
+
+
+def es_search_pointing_ids_from_ref_in_terms(fields, old_link):
+    body = {
+        "query": {
+            "bool": {
+                "should": [
+                    {
+                        "term": {
+                            _key: {"value": old_link}}
+                    }
+                    for _key in fields]
+            }
+        },
+        "fields": [
+            "_id"
+        ]
+    }
+
+    scroll = scan(es, query=body, index="records-*")
+
+    record_uuids = []
+    for response in scroll:
+        _rec_uuid = response['_id']
+        if _rec_uuid:
+            record_uuids.append(_rec_uuid)
+
+    return record_uuids
+
+
+def find_and_replace_old_link_with_new_link_in_record(record, new_link, old_link):
+    if isinstance(record, dict):
+        for key, value in record.items():
+            if key not in bad_fields:
+                if isinstance(value, dict):
+                    for key_ref in value.keys():
+                        if '$ref' == str(key_ref):
```

Comment:
And I swap: variable on the left and value on the right. Since in this way you can read it more naturally:

*If the variable key_ref has value $ref*, instead of *if the value $ref has(?) the variable key_ref.